himl/main.py

```diff
@@ -31,7 +47,8 @@ def do_run(self, opts):
         config_processor = ConfigProcessor()
         config_processor.process(cwd, opts.path, filters, excluded_keys, opts.enclosing_key, opts.remove_enclosing_key,
                                  opts.output_format, opts.print_data, opts.output_file, opts.skip_interpolation_resolving,
-                                 opts.skip_interpolation_validation, opts.skip_secrets, opts.multi_line_string)
+                                 opts.skip_interpolation_validation, opts.skip_secrets, opts.multi_line_string,
+                                 type_strategies= [(list, [self.strategy_merge_override,"append"]), (dict, ["merge"])] )
```

Comment:
This is actually changing the current behaviour by setting the new strategy as default.
Please add a new argument "use-merge-list-strategy" and add this if that is set else keep the old behaviour

Reply:
@oldgromit what I meant is that `himl/main.py` is not really just an example but the main entry point in the `himl` binary installed by the python package:
https://github.com/adobe/himl/blob/master/setup.py#L59

So to avoid breaking existing clients using this with the default strategy I suggested you add a new argument here: https://github.com/adobe/himl/blob/master/himl/main.py#L66-L68
`--use-merge-list-strategy` and when only that is set configure the new strategy.

Reply:
I see what you mean. my understanding of main.py as an example is misguided. 
I can definitely add a new command line argument for the strategy. However, the new strategy would be a function that should be implemented and provided by the user, not himl. is that ok we dynamically load the user specified module and pass the function to Merger, if the optional argument is passed?

Reply:
@oldgromit I'm wondering what would be the best way for you override the merging strategy. 

1. Are you using `himl` as a standalone binary? In that case, it might be useful for you to have an argument like @amuraru mentioned, where you can control the merging behavior. The way I see it, the argument could be a `str` and receive one of the 3 values:
`--merge-list-strategy=append/override/prepend` (see [docs](https://deepmerge.readthedocs.io/_/downloads/en/latest/pdf/)), where _append_ is the default:

```python
parser.add_argument('--merge-list-strategy',
                    choices=['append', 'override', 'prepend'],
                    help='...')
```

Then you can run himl like this to achieve your goal:
```python
himl --merge-list-strategy=override ...
```

For this to work, you can [add the argument](https://github.com/adobe/himl/blob/master/himl/main.py#L66-L68) and then update this line of code to something like:
```
type_strategies= [(list, [opts.merge_list_strategy]), (dict, ["merge"])] )
```

In this way, we pass the value received from the user (override/append/prepend) to the deepmerge lib.

PS: Not sure if it's worth adding an argument for the dict (`--merge-dict-strategy=merge/override`) as well, probably not atm.

2. If you are using `himl` as a python module, it looks like you can pass in the desired strategies with the code you've written so far, correct?

Reply:
thanks @costimuraru  for your feedback. you brought up an interesting point I haven't considered. the 3 values (append/override/prepend) are 3 merge strategies deepmerge provides out of box, we should allow users to specify which one they want to use. I've updated the code to offer that flexibility. 

The original goal of my change is to allow a user specified merge strategy that can be passed to himl. a custom function that user has total control of. the example I provided merges/overrides depends on an "id" attribute, user can specify any other combo of fields to uniquely identifying records, they have total control of merging behavior.

I've pushed a new PR to illustrate how user can build their own custom function and himl dynamically loads specified module and function and pass it as the new merge strategy. Currently it only works with list data type.

Regarding #2 you mentioned, yes, if user have access to ConfigProcessor directly, they have a lot of more freedom and can pass in more elaborate merge strategies. 

@amuraru @costimuraru  Please review the latest PR and provide your valuable feedback. I'll update the documentation once PR is approved. Thanks!